### PR TITLE
fix(cli): handle encoded requirements.txt files

### LINKED ISSUE #37: `numerous init` adds a sequence of characters at the end of requirements.txt
Right after I run `numerous init` to initialize my app, the following sequence of characters appears at the end of the _requirements.txt_ file: 瀊湡汥
image
It then causes a problem when deploying the app at the `Build Step 4/7 : Runn pip install -r /app/requirements.txt`.
The solution to deploy the app anyway is to manually remove these characters before running `numerous deploy`, but I thought it was not an intended behavior.

### PR DESCRIPTION
**Pull request by jfeodor**
* Handle `requirements.txt` files encoded with `utf-8`, `utf-16` and `utf-32` regardless
  of endianness, by detecting the corresponding BOMs.
* Read `requirements.txt` files with both `lf` and `crlf` line termination.
* Preserve `crlf` line termination by checking if any lines are terminated by
  `crlf`, and then using `crlf` when re-encoding if any `crlf` line termination was found.

Fixes #37

### CODE REVIEW

File: internal/requirements/requirements.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	return &requirementsTxt{
+		encoder: encoding.NewEncoder(),
+		lines:   lines,
+		crlf:    crlf,
+	}, nil
+}
+
+func (r *requirementsTxt) Write(w io.Writer) error {
```
Comment (mikhailNumerous) [inline internal/requirements/requirements.go:51]: The concatenation can be done in memory, e.g. this way:
``` go
func (r *requirementsTxt) Write(w io.Writer) error {
	ew := r.encoder.Writer(w)

	lineEnding := "\n"
	if r.crlf {
		lineEnding = "\r\n"
	}

	data := strings.Join(r.lines, lineEnding)
	_, err := ew.Write([]byte(data))

	return err
}

```

File: internal/requirements/requirements.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		r.lines = append(r.lines, "")
+	} else {
+		r.lines = append(r.lines, added)
+		r.lines = append(r.lines, "")
+	}
+}
+
+// dropCR drops a terminal \r from the data, and return true.
+func dropCR(data []byte) ([]byte, bool) {
+	if len(data) > 0 && data[len(data)-1] == '\r' {
```
Comment (mikhailNumerous) [inline internal/requirements/requirements.go:94]: Maybe `if bytes.HasSuffix(data, []byte("\r")) {` would be readable.

File: internal/requirements/requirements.go (hunk 1 above)
Comment (jfeodor) [reply internal/requirements/requirements.go:51]: Good point! That's much nicer.

File: internal/requirements/requirements.go (hunk 2 above)
Comment (jfeodor) [reply internal/requirements/requirements.go:94]: True. This is basically just copied from the built-in version.

Comment (jfeodor) [reply internal/requirements/requirements.go]: I changed it to your suggestion, except using `[]byte{'\r'}`

File: internal/requirements/requirements.go (hunk 1 above)
Comment (jfeodor) [reply internal/requirements/requirements.go:51]: I updated the code with your suggestion :smile:

**Review by mikhailNumerous — APPROVED** [review]
Looks good.

**Pull request merged**